runtime/src/main/java/com/flipkart/flux/deploymentunit/ExecutableRegistryPopulator.java

```diff
@@ -0,0 +1,146 @@
+ * which are annotated with {@link com.flipkart.flux.client.model.Task} in Executable Registry for the later execution.
+ * @author shyam.akirala
+ */
+@Singleton
+public class ExecutableRegistryPopulator implements Initializable {
+
+    /** Logger for this class*/
+    private static final Logger LOGGER = LoggerFactory.getLogger(ExecutableRegistryPopulator.class);
+
+    private DeploymentUnitUtil deploymentUnitUtil;
+
+    private ExecutableRegistry executableRegistry;
+
+    private Set<String> routerNames;
+
+    @Inject
+    public ExecutableRegistryPopulator(DeploymentUnitUtil deploymentUnitUtil, ExecutableRegistry executableRegistry) {
+        this.deploymentUnitUtil = deploymentUnitUtil;
+        this.executableRegistry = executableRegistry;
+    }
+
+    @Override
+    public void initialize() {
+        routerNames = new HashSet<String>();
+        List<String> deploymentUnitNames = deploymentUnitUtil.getAllDeploymentUnits();
+
+        //for each deployment unit
+        for(String deploymentUnitName : deploymentUnitNames) {
+            try {
+
+                //create a class loader and get required classes from it
+                URLClassLoader classLoader = deploymentUnitUtil.getClassLoader(deploymentUnitName);
+                Class TaskClass = classLoader.loadClass(Task.class.getCanonicalName());
+                Class InjectorClass = loadClassLoaderInjector(classLoader);
+                Method getClassInstanceMethod = InjectorClass.getMethod("getClassInstance", Class.class);
+                Class guiceModuleClass = classLoader.loadClass("com.google.inject.Module");
+
+                Object injectorClassInstance;
+                String DUModuleClassFQN = String.valueOf(deploymentUnitUtil.getProperties(classLoader).getProperty("guiceModuleClass"));
+
+                //check if user has specified any guice module class name in deployment unit configuration file, if not create an empty injector
+                if(DUModuleClassFQN == null || DUModuleClassFQN.trim().isEmpty() || DUModuleClassFQN.equals("null"))
+                    injectorClassInstance = InjectorClass.newInstance();
+                else {
+                    injectorClassInstance = InjectorClass.getConstructor(guiceModuleClass).newInstance(classLoader.loadClass(DUModuleClassFQN).newInstance());
+                }
+
+                Set<Method> taskMethods = deploymentUnitUtil.getTaskMethods(classLoader);
+                //for every task method found in the deployment unit create an executable and keep it in executable registry
+                for(Method method : taskMethods) {
+                    String taskIdentifier = new MethodId(method).toString();
+                    Annotation taskAnnotation = method.getAnnotationsByType(TaskClass)[0];
+                    Class<? extends Annotation> annotationType = taskAnnotation.annotationType();
+                    long timeout = 1000l; //default timeout
```

Comment:
inject this using configuration. If not, at least use a constant with a todo marker